src/Tasks/Microsoft.NET.Build.Tasks/GenerateResxCodeBehindFile.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public class GenerateResxCodeBehindFile : TaskBase
+    {
+        [Required]
+        public ITaskItem[] ResxFiles { get; set; }
+
+        [Required]
+        public string Language { get; set; }
+
+        [Output]
+        public ITaskItem[] Compile { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            // The algorithm used by this class is mostly borrowed from:
+            // https://github.com/dotnet/roslyn-tools/blob/master/sdks/RepoToolset/tools/GenerateResxSource.csx,
+            // heavily modified for readability.
+
+            List<ITaskItem> outputs = new List<ITaskItem>();
+
+            foreach (ITaskItem item in ResxFiles)
+            {
+                string codeBehindPath = item.GetMetadata("CodeBehindFile");
+                if (string.IsNullOrEmpty(codeBehindPath))
+                    throw new BuildErrorException("Item '{0}' does not have CodeBehindFile metadata", item.ItemSpec);
+
+                try
+                {
+                    if (Language.ToUpperInvariant() == "C#")
+                    {
+                        WriteCSharpCodeBehind(item, codeBehindPath);
+                    }
+                    else if (Language.ToUpperInvariant() == "VB")
+                    {
+                        WriteVisualBasicCodeBehind(item, codeBehindPath);
+                    }
+                    else
+                    {
+                        throw new BuildErrorException($"Unrecognized language '{Language}'");
```

Comment:
Would be any plans to support F# for ResX code-generation?

Reply:
@kant2002 Not in this PR, I'm afraid; I don't know a thing about F#. Sorry about that!